kore/src/Kore/Attribute/Functional.hs

```diff
@@ -25,27 +26,16 @@ Note: This attribute is also used to annotate axioms stating functionality
 constraints.
  -}
 newtype Functional = Functional { isDeclaredFunctional :: Bool }
-    deriving (GHC.Generic, Eq, Ord, Show)
-
-instance Semigroup Functional where
-    (<>) (Functional a) (Functional b) = Functional (a || b)
-
-instance Monoid Functional where
-    mempty = Functional False
+    deriving (Eq, Ord, Show)
+    deriving (GHC.Generic)
+    deriving anyclass (Hashable, NFData)
+    deriving anyclass (SOP.Generic, SOP.HasDatatypeInfo)
+    deriving anyclass (Debug, Diff)
+    deriving (Semigroup, Monoid) via Monoid.Any
```

Comment:
So both `Functional` and `Any` are newtypes over `Bool`, and you can derive `Functional`'s `Monoid` instance from the one defined on `Any`. This is nice!